src/bo4e/schemata/lowercamelschema.py

```diff
@@ -0,0 +1,108 @@
+"""
+Functions and classes related to case conversions in schemata.
+"""
+import re
+from typing import Pattern
+
+from marshmallow import Schema, pre_load, post_dump
+
+
+def snake_to_lower_camel_case(string: str) -> str:
+    """
+    Converts the string s from snake_case to lowerCamelCase
+    :param string: foo_bar
+    :return: fooBar
+    """
+    # copied from https://marshmallow.readthedocs.io/en/stable/examples.html?highlight=case#inflection-camel-casing-keys
+    parts = iter(string.split("_"))
+    return next(parts) + "".join(i.title() for i in parts)
+
+
+_lower_camel_word_boundary_pattern: Pattern = re.compile("[^A-Z][A-Z]")
+
+
+def lower_camel_to_snake_case(string: str) -> str:
+    """
+    Converts the string s from lowerCamelCase to snake_case
+    :param string: fooBar
+    :return: foo_bar
+    """
+    result = string
+    for lower_upper in _lower_camel_word_boundary_pattern.findall(string):
+        result = result.replace(lower_upper, lower_upper[0] + "_" + lower_upper[1])
+    return result.lower()
+
+
+_upper_camel_word_boundary_pattern: Pattern = re.compile("[A-Z][^A-Z]")
+
+
+def upper_camel_to_snake_case(string: str) -> str:
+    """
+    Converts the string s from UpperCamel to snake_case
+    :param string: FooBar
+    :return: foo_bar
+    """
+    result = string
+    for upper_lower in _upper_camel_word_boundary_pattern.findall(string):
+        result = result.replace(upper_lower, upper_lower[0].lower() + "_" + upper_lower[1])
+    return result
+
+
+def to_snake_case(string: str) -> str:
+    """
+    Converts any string that is either lowerCamel or UpperCamel to snake_case
+    :param string: fooBar or FooBar
+    :return: foo_bar
+    """
+    result = lower_camel_to_snake_case(string)
+    result = upper_camel_to_snake_case(result)
+    return result
+
+
+class SnakeCaseSchema(Schema):
```

Comment:
I think the name of the class does not match so well the purpose of this class.
It serves as a two way converter snake_case <-> camelCase (/ CamelCase?)

```suggestion
class CaseConverterSchema(Schema):
```
or
```suggestion
class CaseConversionSchema(Schema):
```

Reply:
a185f57911b2d53ba2d964afe875a3c990d96f8c